### PR TITLE
Extend RTO1000 driver

### PR DESCRIPTION
**Pull request by nyxus**
Hi, 

I've fixed problems, extended and improved the RTO1000 driver. The main extension is the measurement submodule. This allows the sure to setup, start and clear all 8 measurements.  I've tested this PR using an RTO1024 on my desk. 

Changes proposed in this pull request:
### Driver
- Added submodules and parameters to do a measurement (on all 8 measurements)
- Added error count and error next parameters 
- Added trigger modes
- Added functions check readout if the scope is running, triggered and acquiring data.
- Changed the opc and stop_opc functions so they do not result in errors 
- Added a warning in docstring of the High definition mode

### Driver example
- Added basic example on how to use measurement 
- Added useful function to kind of automate the process of measurements

@WilliamHPNielsen @AdriaanRol

### CODE REVIEW

File: qcodes/instrument_drivers/rohde_schwarz/RTO1000.py [hunk 1]
```diff
@@ -129,6 +129,146 @@ def get_raw(self):
...
+                        # Histogram measurements
+                        'WCOunt',       'WSAMples',     'HSAMples',     'HPEak',
+                        'PEAK',         'UPEakvalue',   'LPEakvalue',   'HMAXimum',
+                        'HMINimum',     'MEDian',       'MAXMin',       'HMEan',
+                        'HSTDdev',      'M1STddev',     'M2STddev',     'M3STddev',
+                        'MKPositive',   'MKNegative'
+                        )
+
+        self.add_parameter('enable',
+                           label='Measurement {} enable'.format(meas_nr),
```
Comment (sohailc) [inline qcodes/instrument_drivers/rohde_schwarz/RTO1000.py]: Do you know you can do this now: 
```python
label=f'Measurement {meas_nr} enable'
```

File: qcodes/instrument_drivers/rohde_schwarz/RTO1000.py [hunk 2]
```diff
@@ -367,6 +508,18 @@ def __init__(self, name: str, address: str,
                                         'CH4': 'CHAN4',
                                         'EXT': 'EXT'})
 
+        self.add_parameter('trigger_mode',
+                           label='Trigger mode',
+                           set_cmd='TRIGger:MODE {}',
+                           get_cmd='TRIGger1:SOURce?',
+                           val_mapping={'AUTO': 'AUTO',
+                                        'NORMAL': 'NORMal',
```
Comment (sohailc) [inline qcodes/instrument_drivers/rohde_schwarz/RTO1000.py]: Are you sure the command argument is case sensitive? If not, there is no need for a value mapping

File: qcodes/instrument_drivers/rohde_schwarz/RTO1000.py (hunk 1 above)
Comment (nyxus) [reply qcodes/instrument_drivers/rohde_schwarz/RTO1000.py]: I did not know this. It's a nice feature! Updated the driver to this new format.

File: qcodes/instrument_drivers/rohde_schwarz/RTO1000.py (hunk 2 above)
Comment (nyxus) [reply qcodes/instrument_drivers/rohde_schwarz/RTO1000.py]: I've tested on the RTO1024 and the commands are not case sensitive. So change the mapping to a vals so int inputs are validated.

**Comment by nyxus** [comment]
I've made the requested changes, @sohailc can you continue your review?

**Comment by nyxus** [comment]
@WilliamHPNielsen can you review this pr?

**Comment by WilliamHPNielsen** [comment]
@nyxus, great! I noticed that you also added some more functionality. Is the driver done now, or are you going to add more later?

**Comment by nyxus** [comment]
I've made the changes as requested, including expending the example notebook. Also added a couple of other functions to the driver to readout if the scope is running, triggered and acquiring data. These functions are used to check if a measurement is possible. The example notebook also contains a useful function to kind off automate the process of measuring.

**Comment by nyxus** [comment]
@WilliamHPNielsen The driver is done for now. It still misses lots of functions/parameters but I did not need them. I will create new PR(s) if that changes.

**Comment by jenshnielsen** [comment]
@WilliamHPNielsen is this ready to merge?

**Comment by WilliamHPNielsen** [comment]
@jenshnielsen, yes, I believe it is (sorry for slow response).

**Pull request merged**